src/framework/ui/dropdown/dropdown.spec.tsx

```diff
@@ -0,0 +1,262 @@
+    const expectedSelectedOption: DropdownItemType[] = [
+      { text: 'Option 4' },
+      { text: 'Option 32' },
+    ];
+    const onMultiSelectPress = jest.fn();
+    const application: RenderAPI = render(
+      <TestApplication onMultiSelectPress={onMultiSelectPress}/>,
+    );
+
+    fireEvent.press(application.getAllByType(TouchableOpacity)[1]);
+    fireEvent(application.getByTestId('@dropdown-item/Option 4'), 'onChange');
+    fireEvent(application.getByTestId('@dropdown-item/Option 32'), 'onChange');
+    const { selectedOption } = application.getByTestId(dropdownMultiSelectTestId).props;
+
+    expect(stringify(selectedOption)).toBe(stringify(expectedSelectedOption));
+  });
+
+  it('* disabled props checks', () => {
+    const onDropdownPress = jest.fn();
+    const onMultiSelectPress = jest.fn();
+    const application: RenderAPI = render(
+      <TestApplication
+        dropdownDisabled={false}
+        multiSelectDisabled={true}
+        onDropdownPress={onDropdownPress}
+        onMultiSelectPress={onMultiSelectPress}
+      />,
+    );
+
+    fireEvent.press(application.getAllByType(TouchableOpacity)[0]);
+    expect(onDropdownPress).toHaveBeenCalled();
+    fireEvent.press(application.getAllByType(TouchableOpacity)[1]);
+    expect(onMultiSelectPress).toHaveBeenCalledTimes(0);
+  });
+
+  it('* multiSelect unselect works properly', () => {
+    const onMultiSelectPress = jest.fn();
+    const application: RenderAPI = render(
+      <TestApplication onMultiSelectPress={onMultiSelectPress}/>,
+    );
+
+    fireEvent.press(application.getAllByType(TouchableOpacity)[1]);
+    fireEvent(application.getByTestId('@dropdown-item/Option 4'), 'onChange');
+    fireEvent(application.getByTestId('@dropdown-item/Option 4'), 'onChange');
+    const { selectedOption } = application.getByTestId(dropdownMultiSelectTestId).props;
+
+    expect(stringify(selectedOption)).toBe(stringify([]));
+  });
+
+  it('* other props checks', () => {
+    const onDropdownPress = jest.fn();
+    const onDropdownPressIn = jest.fn();
+    const onDropdownPressOut = jest.fn();
+    const onDropdownLongPress = jest.fn();
+    const passedLabel: string = 'Label';
+    const passedPlaceholder: string = 'Placeholder';
+    const application: RenderAPI = render(
+      <TestApplication
+        dropdownLabel={passedLabel}
+        dropdownPlaceholder={passedPlaceholder}
+        onDropdownPress={onDropdownPress}
+        onDropdownPressIn={onDropdownPressIn}
+        onDropdownPressOut={onDropdownPressOut}
+        onDropdownLongPress={onDropdownLongPress}
+      />,
+    );
+
+    fireEvent(application.getByTestId(dropdownTestId), 'pressIn');
+    fireEvent(application.getByTestId(dropdownTestId), 'pressOut');
+    fireEvent(application.getByTestId(dropdownTestId), 'longPress');
+    fireEvent.press(application.getAllByType(TouchableOpacity)[0]);
+
+    const { label, placeholder } = application.getByTestId(dropdownTestId).props;
```

Comment:
This is also incorrect way of testing. Expected behavior is that passed `label` and `placeholder` values are finally rendered somewhere. For example, you can use [getByText](https://callstack.github.io/react-native-testing-library/docs/api-queries#bytext) query and expect `children` `toEqual` `label` for this purpose

Reply:
Let's also split this case into multiple cases for each property

Reply:
refactored